### DISCUSSION TITLE
Optimization works only with specific field components.

### DISCUSSION THREAD

**Discussion by Astrei** [bfieldtools/bfieldtools#51] [Q&A] [answered]
Hi guys, thank you very much for such a great tool!
I tried to play around with the simplest helmet example and it works well:

![BF1](https://user-images.githubusercontent.com/7473110/200838948-16fce660-68ec-4852-b1ab-8fd4b8f25124.jpg)

But this only gives me a solution if I set field terms 0 or 3  and not with 1 and 2:

![BF2](https://user-images.githubusercontent.com/7473110/200842901-fdb6446b-a991-4d97-9d25-f09159d7f3e4.jpg)

I did the trick and rotated the mesh by 90 deg, so mathematically it should be the same task, and then it works:

![BF3](https://user-images.githubusercontent.com/7473110/200839850-ea30b642-944a-471f-baaf-63b16d535f1f.jpg)

![BF4](https://user-images.githubusercontent.com/7473110/200844952-7c703f48-afdc-4536-9193-35c152b884e1.jpg)

Could you give me some advice on what am I doing wrong?

**Comment by rzetter** [comment, marked as answer]
Hi, good question!

The issue seems to be the lines in the example code just underneath the sph component choice. They were "hardcoded" for blm[3] += 1 and need to be modified for other components

```
# Set one specific component to one
blm[3] += 1

sphfield = sphtools.field(target_points, alm, blm, lmax)

target_field = sphfield / np.max(sphfield[:, 2])

target_field[:, 2] = 0
```

The second-to-last line scales the target field to a suitable numerical scale, with maximum values of 1. The last lines simply sets all z-values to zero as for blm[3] they were "ugly" floating point error values on the scale of <1e-16 (this is really not needed). 

By modifying the code as below there shouldn't be any issues

```
# Set one specific component to one
blm[3] += 1

sphfield = sphtools.field(target_points, alm, blm, lmax)

target_field = sphfield / np.max(np.abs(sphfield))
```

**Reply by Astrei** [reply]
Thank you for the quick reply. Now it works!
